### PR TITLE
Major Jetbrains HTTP Client upgrade. Move to BETA

### PR DESCRIPTION
**Pull request by jlengrand**
~~Do not merge, this is meant to be WIP and used for CI~~ Getting ready for merge!

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jlengrand** [comment]
@jlengrand @gcatanese @wing328 FYI!

**Comment by jlengrand** [comment]
Alright all checks have passed, let me know how I can help making this a successful review. All notable changes are in `JetbrainsHttpClientClientCodegen` and `api.mustache`

**Comment by gcatanese** [comment]
@jlengrand @wing328 Looks good!

File: .gitignore [hunk 1]
```diff
@@ -282,3 +282,4 @@ samples/openapi3/client/petstore/go/privatekey.pem
 
 ## OCaml
 samples/client/petstore/ocaml/_build/
+/samples/client/jetbrains/adyen/checkout71/http/client/Apis/http-client.private.env.json
```
Comment (wing328) [inline .gitignore:285]: can you please add `#jetbrains-http-client` (with a line break before that) as this line is not related to ocaml client generator at all?

Comment (wing328) [reply .gitignore:285]: i'll do it after merging the PR. have a nice weekend

Comment (wing328) [reply .gitignore:285]: done
